src/manifest/dbtWorkspaceFolder.ts

```diff
@@ -83,7 +79,34 @@ export class DBTWorkspaceFolder implements Disposable {
       {},
       { numProjects: projectFiles.length },
     );
-    return projectFiles.forEach((uri) => this.registerDBTProject(uri));
+    await Promise.all(
+      projectFiles.map(async (uri) => {
+        await this.registerDBTProject(uri);
+      }),
+    );
+    // Filter projects
+    const packagesInstallPaths =
+      await this.findDBTProjectPackagesInstallPaths();
+    const filteredProjectFiles = projectFiles.filter((uri) => {
+      return packagesInstallPaths.some((path) =>
+        uri.toString().includes("/" + path),
+      );
+    });
+    await Promise.all(
+      filteredProjectFiles.map(async (uri) => {
+        await this.unregisterDBTProject(uri);
+      }),
+    );
+    await Promise.all(
+      this.dbtProjects.map(async (project) => {
+        await project.initializeDBTProject();
+      }),
+    );
+
+    // Exception handling after filtering
+    this.dbtProjects.map((project) => {
+      project.handlePythonBridgeException();
```

Comment:
This code should also be moved here. No need to have a public method and a variable for the exception